### PR TITLE
Updates Exercise README

### PR DESCRIPTION
**Pull request by VoodooLlama**
### Overview
Based on the FED team brainstorming session last week, I'm updating the content here to align with the team's expectations for submissions. 

### Includes:
* Adds preference for component-based frameworks
* Adds preference for ES6+
* Adds preference for pull request submission over other methods
* Tweaks messaging a bit

### Notes:
Would love any and all feedback from the team here, gramatically or otherwise.

### CODE REVIEW

**Review by adamrmoss — APPROVED** [review]
Looks awesome, I have no recommended changes!

**Comment by mdatelle** [comment]
Wording sounds good to me.

**Review by mdatelle — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -13,20 +12,18 @@ The test data is a flat structure with parent ID pointers. You may choose to use
...
-- Feel free to make this component as cool/interesting as you'd like, as long
-  as it meets the technical requirements
-- You may use a JavaScript framework of your choice, or no framework &mdash;
-  the choice is up to you
-- However, you must not rely on a module or library that already provides a
-  similar tree component
-- Your solution should be coded with reusability in mind
-- Submit a working set of code (ZIP, tarball, or link to JSFiddle, Plunker,
-  GitHub pull request, etc.)
+- This exercise should take you a few hours to complete, and you are encouraged to take as much time as you feel is necessary.
```
Comment (thehappycactus) [inline README.md]: `and you are encouraged` vs. `but you are encouraged`?

File: README.md [hunk 2]
```diff
@@ -13,20 +12,18 @@ The test data is a flat structure with parent ID pointers. You may choose to use
...
+- Feel free to make this component as cool/interesting as you'd like, as long as it meets the technical requirements. This is your opportunity to show off your skills and knowledge.
+- We'd prefer to see usage of a modern component-based JavaScript framework of your choice (React, Vue, Angular, etc.), but we will consider vanilla submissions as well
+- You should utilize ES6+ syntax (or Typescript) targeting modern browsers
+- You should avoid using any runtime dependencies which provide a similar tree-like structure
+- Your solution should be coded with reusability and maintainability in mind. We'd like to see what your code would look like in a production-like environment.
```
Comment (thehappycactus) [inline README.md]: IMO this is the sort of thing we should _expect_ from a strong programmer, not something we should have to explicitly tell someone to do.  thoughts?

File: README.md [hunk 3]
```diff
@@ -13,20 +12,18 @@ The test data is a flat structure with parent ID pointers. You may choose to use
...
+- We'd prefer you to submit the exercise as a pull request to this repository, but will also accept a .ZIP archive or link to an online code editor (JSFiddle, etc.)
   - A working, compiled set of code in a single directory that can be opened in a browser
+  - Instructions for running any build or test steps, as well as any clarification or justification you'd like to provide
```
Comment (thehappycactus) [inline README.md]: grammar nitpick: comma splice

File: README.md [hunk 4]
```diff
@@ -13,20 +12,18 @@ The test data is a flat structure with parent ID pointers. You may choose to use
...
-  as it meets the technical requirements
-- You may use a JavaScript framework of your choice, or no framework &mdash;
-  the choice is up to you
-- However, you must not rely on a module or library that already provides a
-  similar tree component
-- Your solution should be coded with reusability in mind
-- Submit a working set of code (ZIP, tarball, or link to JSFiddle, Plunker,
-  GitHub pull request, etc.)
+- This exercise should take you a few hours to complete, and you are encouraged to take as much time as you feel is necessary.
+- Feel free to make this component as cool/interesting as you'd like, as long as it meets the technical requirements. This is your opportunity to show off your skills and knowledge.
```
Comment (thehappycactus) [inline README.md]: grammar nitpick: comma splice

File: README.md (hunk 2 above)
Comment (VoodooLlama) [reply README.md]: I think it helps frame the exercise as a "real world" example of code, as opposed to a one-off toy exercise. Maybe remove the first sentence here, re-word the second sentence a bit?

Comment (thehappycactus) [reply README.md]: that's fair.  how about "Keep in mind this should reflect what your code would look like in a production-type environment"?

**Review by RainbowPawz — APPROVED** [review]
Looks good to me!

**Review by adamrmoss — APPROVED** [review]
Now it's even better!

**Review by thehappycactus — APPROVED** [review]

**Comment by VoodooLlama** [comment]
Going ahead and merging this based on reviews. Thanks everyone!

**Pull request merged**